### PR TITLE
Don't open the SQR submission form if the token is invalid

### PR DESCRIPTION
**Pull request by DadImScared**
https://trello.com/c/lLI19ZXG/232-dont-open-the-sqr-submission-form-if-the-token-is-invalid

### CODE REVIEW

File: frontend/src/views/SQR/Form.vue [hunk 1]
```diff
@@ -348,6 +356,27 @@ export default class Form extends Vue {
...
+      .database()
+      .ref(`SQR/allotments`)
+      .orderByChild("token")
+      .equalTo(token)
+      .once("value")).val();
+    const sqrStatus = _.get<string>(response, `${fileName}.status`, "");
+    if (!sqrStatus) {
+      this.allotmentError =
+        "This allotment is not valid, please contact coordinator.";
+    } else if (!this.isCoordinator && sqrStatus.toLowerCase() === "done") {
```
Comment (anantakrishna) [inline frontend/src/views/SQR/Form.vue]: I'm not sure it's worth adding `isCoordinator` property that is used only once in the code. Why not to query the user claims here directly? Or even better, add `isCoordinator` to the `user` store module and map it from there.

File: frontend/src/views/SQR/Form.vue [hunk 2]
```diff
@@ -348,6 +356,27 @@ export default class Form extends Vue {
...
+    const {
+      params: { fileName, token }
+    } = this.$route;
+    const response = (await firebase
+      .database()
+      .ref(`SQR/allotments`)
+      .orderByChild("token")
+      .equalTo(token)
+      .once("value")).val();
+    const sqrStatus = _.get<string>(response, `${fileName}.status`, "");
```
Comment (anantakrishna) [inline frontend/src/views/SQR/Form.vue]:
```suggestion
    const allotmentStatus = _.get<string>(response, `${fileName}.status`, "");
```

File: frontend/src/views/SQR/Form.vue [hunk 3]
```diff
@@ -381,8 +410,11 @@ export default class Form extends Vue {
   }
 
   async mounted() {
-    this.canSubmit = true;
-    this.getSavedData();
+    await this.checkIsCoordinator();
+    await this.canSubmitForm();
```
Comment (anantakrishna) [inline frontend/src/views/SQR/Form.vue]: Usually the methods should be named in the form “Do this and that”. `canSubmitForm` looks more like a calculated property. Consider renaming or refactoring this.

File: frontend/src/store/modules/user.ts [hunk 4]
```diff
@@ -18,10 +18,14 @@ export default {
...
-          query: { redirect: router.currentRoute.fullPath }
-        });
+        router
+          .push({
+            path: "/login",
+            query: { redirect: router.currentRoute.fullPath }
+          })
+          .catch(() => {
+            // prevent uncaught promise error
+          });
```
Comment (anantakrishna) [inline frontend/src/store/modules/user.ts]: From the perspective of the task, this change looks irrelevant. Is it for fixing http://prntscr.com/prg5qm?

File: frontend/src/views/SQR/Form.vue (hunk 1 above)
Comment (DadImScared) [reply frontend/src/views/SQR/Form.vue]: Where would the initial `isCoordinator` be set in the vuex store? Maybe the vuex commit that runs on [auth state change](https://github.com/PureBhaktiArchive/audioseva/blob/master/frontend/src/main.ts#L59) can be replaced with an action that sets the currentUser and does additional stuff like setting isCoordinator?

File: frontend/src/store/modules/user.ts (hunk 4 above)
Comment (DadImScared) [reply frontend/src/store/modules/user.ts]: Yes

File: frontend/src/views/SQR/Form.vue (hunk 1 above)
Comment (anantakrishna) [reply frontend/src/views/SQR/Form.vue]: ok, let's leave it as of now, but please consider refactoring this whenever you have time. I have added it as a point to https://trello.com/c/vKAD3ViL/17-load-user-info-from-the-routes.

File: frontend/src/views/SQR/Form.vue [hunk 5]
```diff
@@ -312,6 +310,8 @@ export default class Form extends Vue {
   };
   submitSuccess = false;
   formHasError = false;
+  isCoordinator = false;
+  allotmentError = "";
```
Comment (anantakrishna) [inline frontend/src/views/SQR/Form.vue]: As this property is used not only for the allotment errors, I'd name it more generic. For example,
```suggestion
  errorMessage = "";
```
